catalog/plugin/catalog-plugin-metacard-validation/src/main/java/ddf/catalog/metacard/validation/MetacardValidityMarkerPlugin.java

```diff
@@ -0,0 +1,142 @@
+
+        // Initialize empty list of metacards to allow through
+        List<Metacard> returnMetacards = new ArrayList<Metacard>();
+        // Loop all metacards in request
+        for (Metacard metacard : input.getMetacards()) {
+            MetacardImpl newMetacard = new MetacardImpl(metacard);
+            List<Serializable> validationWarnings = new LinkedList<>();
+            List<Serializable> validationErrors = new LinkedList<>();
+            // Default to allowing metacard through
+            Boolean toReturn = true;
+            // Run metacard through each validator
+            for (MetacardValidator metacardValidator : metacardValidators) {
+                try {
+                    // Attempt validation
+                    metacardValidator.validate(metacard);
+                } catch (ValidationException e) {
+                    if (metacardValidator instanceof Describable) {
+                        // If validator is not explicitly turned on by admin, set invalid and allow through
+                        if (checkEnforcedMetacardValidators(metacardValidator)) {
+                            // Check for warnings and errors
+                            if (!e.getErrors().isEmpty()) {
+                                validationErrors.add(((Describable) metacardValidator).getId());
+                            }
+                            if (!e.getWarnings().isEmpty()) {
+                                validationWarnings.add(((Describable) metacardValidator).getId());
+                            }
+                        } else {
+                            // If validator is explicitly turned on, break out and do not include in the
+                            // list of metacards that will be allowed through.
+                            toReturn = false;
+                            break;
+                        }
+                    } else {
+                        // TODO: Do we want validator plugin to fail when a failed validator doesn't implement describable?
+                        // We won't have any good information about the validator to log or put in the 'validationErrors'
+                        // or 'validationWarnings,' but we could silently fail and not ingest the metacard at all instead.
+                        LOGGER.warn("Metacard validators must implement Describable.");
+                        throw new StopProcessingException(
+                                "Metacard validators must implement Describable.");
+                    }
+                }
+            }
+            if (toReturn) {
+                newMetacard
+                        .setAttribute(new AttributeImpl(VALIDATION_WARNINGS, validationWarnings));
+                newMetacard.setAttribute(new AttributeImpl(VALIDATION_ERRORS, validationErrors));
+                returnMetacards.add(newMetacard);
+            }
+        }
+        return new CreateRequestImpl(returnMetacards);
```

Comment:
This constructor will not include the `input` properties.  You should use the `public CreateRequestImpl(List<Metacard> metacards, Map<String, Serializable> properties)` constructor instead.